invisible_cities/cities/beersheba_test.py

```diff
@@ -0,0 +1,185 @@
+import os
+import numpy  as np
+import tables as tb
+import pandas as pd
+
+from pytest                    import mark
+
+from .. core.system_of_units_c import units
+from .. core.configure         import configure
+from .. core.configure         import all         as all_events
+from .. io                     import dst_io      as dio
+from .  beersheba              import beersheba
+from .  beersheba              import create_deconvolution_df
+from .  beersheba              import distribute_energy
+from .  beersheba              import CutType
+from .. evm.event_model        import HitEnergy
+from .. core.testing_utils     import assert_dataframes_close
+from .. core.testing_utils     import assert_tables_equality
+
+
+def test_create_deconvolution_df(ICDATADIR):
+    true_in  = os.path.join(ICDATADIR, "exact_Kr_deconvolution_with_MC.h5")
+    true_dst = dio.load_dst(true_in, 'DECO', 'Events')
+    ecut = 1e-2
+    new_dst  = pd.concat([create_deconvolution_df(t, t.E.values, (t.X.values, t.Y.values, t.Z.values),
+                                                  CutType.abs, ecut, 3) for _, t in true_dst.groupby('event')])
```

Comment:
Yup, otherwise the selection inside `create_deconvolution_df` won't work properly and return nan values here:

```    
df['X']     = pos[0][sel_deconv]
df['Y']     = pos[1][sel_deconv]
```

as `df` indexes do not  match `pos[sel_deconv]` (with pos equivalent to (t.X, t.Y)). Dunno if there is an easy workaround but the function is thought to work with arrays as an input.

Reply:
You could use `np.asarray` within `create_deconvolution_df` whenever you need to. This will move the ugliness to the guts of the function. Not sure if it is worth it, though...